### DISCUSSION TITLE
Usage of type = "terminal" in alpha.nvim

### DISCUSSION THREAD

**Discussion by brneor** [LunarVim/LunarVim#3787] [Q&A]
I'm having a hard time trying to set a dashboard header using `type = "terminal"` option of alpha.nvim.

I'm used to overriding `header` and `footer` of Lunarvim's dashboard as you can se [here](https://gitlab.com/brneor/lvim-config/-/blob/master/lua/dashboard.lua).

The problem is when I try to use `alpha.term` on the header instead of simple strings I face two different issues:

- The result of my command shows on top of sections `buttons` and `footer` as you can see on next screenshots

![image](https://user-images.githubusercontent.com/5962898/215108719-1aa08cd6-c01a-4fe7-a90e-9fd541c141d1.png)

Smaller terminal height so you can see that `buttons` and `footer` are still there:
![image](https://user-images.githubusercontent.com/5962898/215108583-8113edca-b999-497e-9c7f-5d91a2abbca9.png)

- OOOOR I try to override `lvim.builtin.alpha.dashboard.config.layout` with
```lua
lvim.builtin.alpha.dashboard.config.layout = {
	{ type = "padding", val = 1 },
	lvim.builtin.alpha.dashboard.section.header,
	{ type = "padding", val = height + 8 },
	lvim.builtin.alpha.dashboard.section.buttons,
	{ type = "padding", val = 1 },
	lvim.builtin.alpha.dashboard.section.footer,
}
```

And I get the error 
![image](https://user-images.githubusercontent.com/5962898/215109137-35be4bb1-8886-4740-ae22-7182691af8ba.png)

If I remove the `buttons` from `layout` override everything works just fine:

![image](https://user-images.githubusercontent.com/5962898/215109406-59955a87-27ec-4fcb-8115-1017ccaa5f6c.png)

But I really want the buttons......

What am I missing here? What can be done to support `alpha.term`?

**Comment by brneor** [comment]
@LostNeophyte can you enlighten me in this one?

**Comment by mcursa-jwt** [comment]
hi @brneor, i was facing the same issue in my configs as well, and it was due to the calling of `vim.cmd.AlphaRedraw` in an autocommand that was called after `alpha.setup(config)`.

**Reply by brneor** [reply]
Ty for your help but since then I moved to AstroNvim, specially after [this comment](https://github.com/LunarVim/LunarVim/discussions/4518#discussioncomment-8963843).

**Reply by mcursa-jwt** [reply]
No worries, since alpha-nvim is pretty portable it might still apply to other nvim frameworks :)
